### DISCUSSION TITLE
Insertion of an equation in FDS linux code

### DISCUSSION THREAD

**Discussion by Sun0036** [firemodels/fds#12798] [Q&A]
Hello group members!

I have an equation which will calculate the mass of PM2.5 in kg: 
'PM2.5 [kg pollu] = (Fuel mass @ t– fuel mass @ t-1) [kg fuel] x Burning efficiency x emission factor [ kg pollu/kg fuel]'

We would like to extract the output data of the  (Fuel mass @ t– fuel mass @ t-1) [kg fuel] from fds output excel, then we multiply the Burning efficiency and emission factor, finally, we'are going to obtain the PM2.5. 

If i want to insert this equation into FDS Linux code, how could i do? I will begin from which file of '.f90'?

Thank you! 

![Capture d'écran 2024-04-14 211801](https://github.com/firemodels/fds/assets/162191311/d10e9b11-7286-4931-bb60-e5a62ff9ae53)
![Capture d'écran 2024-04-14 212110](https://github.com/firemodels/fds/assets/162191311/960908d7-830b-4617-a8c1-b2e0a8cce346)

**Comment by drjfloyd** [comment]
If all you want to do is have a fixed yield of a species (i.e. x kg of PM2.5 per kg of fuel burned), you don't need to modify the source code. Just define a REAC that makes some species you define. You could for example define a copy of the SOOT species and call it SOOT_PM2p5 and then have a REAC that makes SOOT and SOOT_PM2p5 for whatever yields you need.  See the User's Guide chapters on defining species and combustion reactions.

**Reply by Sun0036** [reply]
Thank you for you answer.

 'Burning efficiency x emission factor' are values that we give,  but i think that FDS dosen't have the capacity to calculate the (Fuel mass @ t– fuel mass @ t-1) [kg fuel] during the simulation. Due to this situation, we need to modify the source code?

**Comment by drjfloyd** [comment]
Heat release rate = (Fuel mass (t) - Fuel (t-1) )/ (t - (t-1)) * Heat of combustion.

**Comment by drjfloyd** [comment]
The value of burning efficiency x emission factor would be the yield

**Comment by Sun0036** [comment]
Ok, thank you for the response! I will try it.
